src/liballoc/str.rs

```diff
@@ -418,6 +423,7 @@ impl str {
     ///
     /// assert_eq!(boxed_str.into_string(), string);
     /// ```
+    #[must_use]
     #[stable(feature = "box_str", since = "1.4.0")]
     #[inline]
     pub fn into_string(self: Box<str>) -> String {
```

Comment:
This one feels insufficiently justified.  It's not allocating.  It consumes the input so if you didn't need it you'll probably know.  It does almost nothing, so will probably disappear on its own if unused.

I agree that it _could_ be `must_use`, but as I understand it the goal isn't to put it everywhere possible.

Reply:
I don't understand why we should apply so hardly-defined rules to where `must_use` should be used and where it shouldn't. I suggest the following reasoning: if the function call doesn't make sense without consuming the result, it should be marked with `must_use`.

> It does almost nothing, so will probably disappear on its own if unused.

Why does Rust warn about unused `use`s then? They do nothing and just "disappear" completely if not used.